Sources/LeafKit/DocumentLoader.swift

```diff
@@ -1,292 +0,0 @@
-                
-                // the result here is all dependencies that are not
-                // already loaded into the system, attempt to fetch from disk
-                let missing = allUnresolvedDependencies.filter { dependency in
-                    // we've already loaded the dependency
-                    // but haven't been able to resolve it,
-                    // continue up tree to see which loads are missing
-                    return !unresolvedDocuments.contains(dependency)
-                        // this dependency is ready and fully resolved,
-                        // there must be other dependencies
-                        // to resolve before this dependency can work
-                        && !fullyResolved.contains(dependency)
-                }
-                
-                // TODO: Would be nice to detect circular or unresolvable dependencies here
-                guard !missing.isEmpty else {
-                    unresolved = [:]
-                    waitingToRetry.forEach { stillRaw in
-                        unresolved[stillRaw.name] = stillRaw
-                    }
-                    return
-                }
-                
-                // load missing dependencies from files
-                let fresh = try Set(missing).map { name -> UnresolvedDocument in
-                    let raw = try fileAccess.load(name: name)
-                    return try parse(name: name, raw: raw)
-                }
-                
-                
-                // put fresh in front, since they're known dependencies
-                start = fresh + waitingToRetry
-                drain = start
-                waitingToRetry = []
-            } else {
-                // still some unresolved, let's try another pass
-                start = waitingToRetry
-                drain = start
-                waitingToRetry = []
-            }
-        }
-        
-        guard waitingToRetry.isEmpty else { fatalError("can only break when empty") }
-    }
-    
-    private func ready(next doc: UnresolvedDocument) throws {
-        precondition(canSatisfyAllDependenciesFor(doc: doc))
-        
-        var processed: [Syntax] = []
-        try doc.raw.forEach { syntax in
-            if case .extend(let e) = syntax {
-                let base = try load(e.key)
-                let extended = e.extend(base: base.ast)
-                processed += extended
-            } else {
-                processed.append(syntax)
-            }
-        }
-        
-        let new = try ResolvedDocument(name: doc.name, ast: processed)
-        resolved[new.name] = new
-    }
-    
-    private func canSatisfyAllDependenciesFor(doc: UnresolvedDocument) -> Bool {
-        // no deps, easily satisfy
-        return doc.unresolvedDependencies.isEmpty
-            // see if all dependencies necessary have already been compiled
-            || doc.unresolvedDependencies.allSatisfy(resolved.keys.contains)
-    }
-}
```

Comment:
DocumentLoader.swift renamed to LeafAST.swift - changes to specific functions noted in that file